homeassistant/components/climacell/weather.py

```diff
@@ -0,0 +1,305 @@
+            self._get_cc_value(self.coordinator.data[CURRENT], CC_ATTR_CONDITION),
+            is_up(self.hass),
+        )
+
+    @property
+    def visibility(self):
+        """Return the visibility."""
+        visibility = self._get_cc_value(
+            self.coordinator.data[CURRENT], CC_ATTR_VISIBILITY
+        )
+        if self.hass.config.units.is_metric and visibility:
+            return distance_convert(visibility, LENGTH_MILES, LENGTH_KILOMETERS)
+        return visibility
+
+    @property
+    def forecast(self):
+        """Return the forecast."""
+        # Check if forecasts are available
+        if self.coordinator.data[FORECASTS].get(self.forecast_type):
+            forecasts = []
+
+            # Set default values (in cases where keys don't exist), None will be
+            # returned. Override properties per forecast type as needed
+            for forecast in self.coordinator.data[FORECASTS][self.forecast_type]:
+                timestamp = self._get_cc_value(forecast, CC_ATTR_TIMESTAMP)
+                use_datetime = True
+                condition = self._get_cc_value(forecast, CC_ATTR_CONDITION)
+                precipitation = self._get_cc_value(forecast, CC_ATTR_PRECIPITATION)
+                precipitation_probability = self._get_cc_value(
+                    forecast, CC_ATTR_PRECIPITATION_PROBABILITY
+                )
+                temp = self._get_cc_value(forecast, CC_ATTR_TEMPERATURE)
+                temp_low = None
+                wind_direction = self._get_cc_value(forecast, CC_ATTR_WIND_DIRECTION)
+                wind_speed = self._get_cc_value(forecast, CC_ATTR_WIND_SPEED)
+
+                if self.forecast_type == DAILY:
+                    use_datetime = False
+                    precipitation = self._get_cc_value(
+                        forecast, CC_ATTR_PRECIPITATION_DAILY
+                    )
+                    temp = next(
+                        (
+                            self._get_cc_value(item, CC_ATTR_TEMPERATURE_HIGH)
+                            for item in forecast[CC_ATTR_TEMPERATURE]
+                            if "max" in item
+                        ),
+                        temp,
+                    )
+                    temp_low = next(
+                        (
+                            self._get_cc_value(item, CC_ATTR_TEMPERATURE_LOW)
+                            for item in forecast[CC_ATTR_TEMPERATURE]
+                            if "max" in item
+                        ),
+                        temp_low,
+                    )
+                elif self.forecast_type == NOWCAST:
+                    # Precipitation is forecasted in CONF_TIMESTEP increments but in a
+                    # per hour rate, so value needs to be converted to an amount.
+                    precipitation = (
+                        precipitation / 60 * self._config_entry.options[CONF_TIMESTEP]
+                        if precipitation
```

Comment:
We prefer to not use ternary expression if the statement spans more than one line.